app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt

```diff
@@ -5,53 +5,230 @@
+                .fillMaxWidth()
+        ) {
+            if (!message.data.title.isNullOrBlank()) {
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                ) {
+
+                    SectionHeader(
+                        text = message.data.title!!,
+                        modifier = Modifier
+                            .weight(1f)
+                            .padding(bottom = 10.dp)
+                    )
+
+                    IconButton(
+                        modifier = Modifier.size(20.dp),
+                        onClick = onMessageClosedClicked
+                    ) {
+                        Icon(
+                            painter = painterResource(R.drawable.ic_close),
+                            contentDescription = stringResource(
+                                R.string.remove_home_collection_placeholder_content_description
+                            ),
+                            tint = FirefoxTheme.colors.iconPrimary
+                        )
+                    }
+                }
+                PrimaryText(
+                    text = message.data.text,
+                    modifier = Modifier.fillMaxWidth(),
+                    fontSize = 14.sp
+                )
+            } else {
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                ) {
+                    PrimaryText(
+                        text = message.data.text,
+                        modifier = Modifier.weight(1f),
+                        fontSize = 14.sp
+                    )
+                    IconButton(
+                        modifier = Modifier.size(24.dp),
+                        onClick = onMessageClosedClicked
+                    ) {
+                        Icon(
+                            painter = painterResource(R.drawable.ic_close),
+                            contentDescription = stringResource(
+                                R.string.remove_home_collection_placeholder_content_description
+                            ),
+                            tint = FirefoxTheme.colors.iconPrimary
+                        )
+                    }
+                }
             }
-        }
+            if (!message.data.buttonLabel.isNullOrBlank()) {
+                Spacer(modifier = Modifier.height(16.dp))
 
-        binding.close.apply {
-            increaseTapArea(CLOSE_BUTTON_EXTRA_DPS)
-            setOnClickListener {
-                interactor.onMessageClosedClicked(message)
+                MessageCardButton(
+                    onClick = onMessageClicked
+                )
             }
         }
-        interactor.onMessageDisplayed(message)
     }
+}
 
-    companion object {
-        internal const val LAYOUT_ID = R.layout.nimbus_message_card
-        private const val CLOSE_BUTTON_EXTRA_DPS = 38
+/**
+ * Message Card Button.
+ *
+ * @param onClick Invoked when user clicks on button.
+ */
+
+@Suppress("LongMethod")
```

Comment:
nit unnecessary new line, and is this actually a LongMethod?

Reply:
It said while running detekt that it is a long method.

Reply:
It only appears like MessageCardPlaceHolder is a LongMethod from my point of view. MessageCardButton is less than 60 lines.